### PR TITLE
Localization

### PR DESCRIPTION
**Pull request by krishnabvkr**
Added support for multiple locales.
Updated test cases.

### CODE REVIEW

File: wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala [hunk 1]
```diff
@@ -38,13 +39,13 @@ import java.util.{ResourceBundle, Locale}
   */
 trait LocalizedString {
   /** get the message w/o formatting */
-  def raw(msg: String)(implicit locale: Locale=Locale.getDefault, context:String="messages"): String = {
-    val bundle = ResourceBundle.getBundle(context, locale, UTF8BundleControl)
-    bundle.getString(msg)
+  def raw(msg: String)(implicit locale: Locale = Locale.getDefault, context: String = "messages", fallbackLocales: List[Locale] = Nil): (String, Locale) = {
+    val bundle = ResourceBundle.getBundle(context, locale, new UTF8BundleControl(fallbackLocales.to[Queue]))
+    (bundle.getString(msg), bundle.getLocale)
```
Comment (AlexanderNull) [inline wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala:44]: This feels a bit overloaded now. If there's a need to get the locale that a lookup results in then there should be one public method for looking up the bundle, and one public method for getting the string from that bundle

Comment (AlexanderNull) [reply wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala:44]: That's assuming some external consumer needs to lookup which bundle was used. If this new lookup functionality is only intended to be used within the internals of LocalizableString.localize then the lookup can be private to the current scope. Separating out this concern however frees us up to keep the existing interface of LocalizedString the same as current live version while isolating all new changes into the new case class.

File: wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala [hunk 2]
```diff
@@ -90,4 +96,20 @@ private[utils] object UTF8BundleControl extends ResourceBundle.Control {
...
+}
+
+/**
+ * Support for List of Locale in Localization
+ * It accepts ordered List of locales
+ * returns Localized value in highest locale that App supports
+ * */
+case class LocalizableString(key: String, args: List[Any] = Nil) {
+
+  def localize(locales: List[Locale] = List(Locale.getDefault), context: String = "messages"): String = {
```
Comment (AlexanderNull) [inline wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala:108]: This applies to line 106 as well: Try to get as generic of a type as possible on public interfaces. In most cases there's no reason why you need a List over some other Seq so if you change the interface to allow Seq instead it opens up consumers to passing in a multitude of collections.

Comment (AlexanderNull) [reply wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala:108]: I almost feel like context should be implicit or at least a separate params list from localize. I doubt we'll get enough time to fully plan out consuming project structure here though.

File: wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala (hunk 1 above)
Comment (krishnabvkr) [reply wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala]: The lookup functionality is intended to use in LocalizableString.localize only. 
Added private function to get resource bundle in LocalizableString and using that for lookup in localize. Kept the existing interface as is.

File: wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala (hunk 2 above)
Comment (krishnabvkr) [reply wookiee-core/src/main/scala/com/webtrends/harness/utils/LocalizedString.scala]: It is really make sense, Updated from List to Seq. 
separated context from params list and made it as implicit.

**Pull request merged**